src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,148 @@
+import { mount, shallow, render } from 'enzyme'
+import React from 'react'
+import { Switch, MemoryRouter } from 'react-router-dom'
+import { Provider } from 'react-redux'
+import { I18nextProvider } from 'react-i18next'
+import { StylesProvider, ThemeProvider } from '@material-ui/styles'
+
+import { getMockedStore } from '@mocks/state.mock'
+import i18nMock from '@mocks/i18n.mock'
+import { routeMock } from '@mocks/route.mock'
+
+import { App, AppContainer } from '@src/App'
+import Dashboard from '@pages/Dashboard'
+import Notification from '@components/Notification'
+import RouteWithLayout from '@components/RouteWithLayout'
+import Login from '@pages/Login'
+import Logout from '@pages/Logout'
+import NotFoundPage from '@pages/NotFoundPage/NotFoundPage'
+
+const props = {
+  authChecked: true,
+  isAuthenticated: true,
+  checkAuthentication: jest.fn(),
+  ...routeMock({}),
+}
+
+describe('<App />', () => {
+  describe('how to wrap AppContainer', () => {
+    it('should works', () => {
+      const wrapper = mount(
+        <Provider store={getMockedStore({})}>
+          <I18nextProvider i18n={i18nMock}>
+            <MemoryRouter>
+              <AppContainer {...props} />
+            </MemoryRouter>
+          </I18nextProvider>
+        </Provider>,
+      )
+      expect(wrapper).toBeDefined()
+    })
+  })
+
+  describe('App component without wrappers', () => {
+    it('should match the snapshot', () => {
+      const wrapper = shallow(<App {...props} />)
+      expect(wrapper).toMatchSnapshot()
+    })
+  })
+
+  it('should contain specific components if user is logged in via shallow', () => {
+    const wrapper = shallow(<App {...props} />)
+    expect(wrapper.find('ThemeProvider')).toHaveLength(1)
+    expect(wrapper.find('Suspense')).toHaveLength(1)
+    expect(wrapper.find('withI18nextTranslation(Connect(Notification))')).toHaveLength(1)
+    expect(wrapper.find('Switch')).toHaveLength(1)
+    expect(wrapper.find('RouteWithLayout')).toHaveLength(6)
+    expect(wrapper.find('Redirect')).toHaveLength(3)
+  })
+
+  it('should contain specific components if the user is not logged in via shallow', () => {
+    const calculatedProps = {
+      ...props,
+      isAuthenticated: false,
+    }
+
+    const wrapper = shallow(<App {...calculatedProps} />)
+    expect(wrapper.find('ThemeProvider')).toHaveLength(1)
```

Comment:
prefer search by Component rather than string name of component

Reply:
I intentionally show 2 options. Thanks.